File: sokkou-deck/src/banned.rs
pub const BANNED_WORDS: &[&str] = &[
    "失格",
    "クソ",
    "無能",
    "諦めろ",
    "ダメ人間",
    "お前",
];

pub fn find(s: &str) -> Option<&'static str> {
    BANNED_WORDS.iter().find(|w| s.contains(*w)).copied()
}

pub fn audit_all<'a, I: IntoIterator<Item = &'a str>>(strings: I) -> Vec<(String, &'static str)> {
    strings
        .into_iter()
        .filter_map(|s| find(s).map(|w| (s.to_string(), w)))
        .collect()
}

File: kisanae-chou/kisanae_chou/banned.py
BANNED_WORDS = [
    "神 アニメ",
    "神アニメ",
    "クソ アニメ",
    "クソアニメ",
    "神 回",
    "神回",
    "ランキング",
    "TOP 5",
    "TOP 10",
    "最強",
    "お前",
    "ダメ",
]


def find_banned(s: str):
    for w in BANNED_WORDS:
        if w in s:
            return w
    return None


def audit_all(strings):
    return [(s, find_banned(s)) for s in strings if find_banned(s)]
